### DISCUSSION TITLE
Creating new entities somehow broken?

### DISCUSSION THREAD

**Discussion by Alex9779** [zachowj/node-red-contrib-home-assistant-websocket#997] [Q&A] [answered]
Hi dear community,
I don't know exactly after which update, I do not create new entites from Node-Red regularly but currently using HA 2023.7.2 and HA websocket 0.55.1 with the integration from HACS 1.32.1 it does not work as it did before.
I was using Node-Red and all the stuff a long time so I have several very "old" entities gone through multiple migrations I think.
For example I have no "device" set in my entities.
But now creating new entities does not really work if I just enter a "name". I am sure this did work before. I do not get any error messages, at least not at the obvious places like when deploying in Node-Red or HA logs or the logs of the Node-Red addon.
Only if I enter a "friendly name" then I get the entity in HA. Somehow I have the feeling it has been there longer because when I did that to one I created a few days ago but came over it, it had historical data.
The problem is that "name" seems no longer to set the entity ID in HA but instead created some random ID and I only find my entity if I give it a friendly name.
Is that some kind of new thing? Is this the way now that the ID is not configurable anymore and I just have to work with friendly names?

Any help appreciated...

~Alexander

**Comment by zachowj** [comment, marked as answer]
The first time an entity config is deployed in Node-RED, it will attempt to create an entity with the ID based off the text in the friendly name field. If no text is entered, then it will use the node ID, the random ID you've seen, as the `entity_id` in Home Assistant. If you forget to enter a friendly name the first time you deploy, edit the `entity_id` in Home Assistant.

It has been this way since the first version of the custom component.

**Reply by Alex9779** [reply]
Thanks for clarification, no idea why I did it this way this time, I added new sensors a while ago and entered the friendly name and all of the old ones have one too... My bad...
